### ISSUE TITLE
make install also installs the tests

### ISSUE THREAD

**Issue by zoe1337** [intel/mptcpd#72]
**Describe the bug**
`make DESTDIR="/wherever/" install` also copies tests

**To Reproduce**
`./configure --prefix=/usr`
`make`
`make DESTDIR="/wherever/" install`

this creates the $PWD directory tree and tests, like `/home/zoe/Downloads/mptcpdpkg/pkg/mptcpd/home/zoe/Downloads/mptcpdpkg/src/mptcpd-0.4/tests` in my case. (here DESTDIR was "/home/zoe/Downloads/mptcpdpkg/pkg/mptcpd/"

**Expected behavior**
tests should not be installed, and especially not under /prefix/$build_pwd

**Desktop:**
 - Linux Kernel: 5.6.14-arch1-1
 - Compiler: gcc version 10.1.0

**Additional context**
an arch PKGBUILD:
```
pkgname=mptcpd
pkgver=0.4
pkgrel=1
pkgdesc="Multipath TCP daemon"
url="https://github.com/intel/mptcpd"
arch=('x86_64')
license=('GPL2' 'BSD')
depends=('ell')
source=("https://github.com/intel/mptcpd/releases/download/v$pkgver/$pkgname-$pkgver.tar.gz")
sha512sums=('1db9fa7182b167d45c44e16dcec051f426039eeeb84dc5b7352819f5b5048e5f5cdc22c567b32054b72bed444869cad36b24c3354e63d671a807010d0b9aab68')

#prepare() {
#       cd "$pkgname-$pkgver"
#    #./bootstrap # only required when building a git clone
#}

build() {
        cd "$pkgname-$pkgver"
    ./configure --prefix=/usr 
    make
}

package() {
        cd "$pkgname-$pkgver"
    make DESTDIR="$pkgdir/" install
    echo "$PWD"
}
```

**Comment by ossama-othman** [comment]
Nice catch!

This looks like an artifact of my attempt to force test plugins to be built and installed in the `mptcpd/tests` directory.  In `tests/Makefile.am`:

```Makefile
## Build a test mptcpd plugin.
##
## "check_LTLIBRARIES" can't be used here since a dynamically loadable
## module will not be built.  A Libtool convenience library would be
## built instead.
##
## Absolute path needed by Libtool during installation.
plugin_adir    = $(abs_builddir)/plugins_a
...
plugin_a_LTLIBRARIES = plugin_three.la plugin_four.la
...
```
The generated `tests/Makefile` ends up prepending `$(DESTDIR)` to the test plugin installation directories, which then causes them to be potentially installed outside of the `mptcpd/tests` directory when `DESTDIR` is set.  Ideally, the test plugins would only be built when running `make check`, but then we run into the problem described in the above `Makefile.am` snippet.  I'll see what I can do about this problem.  Thanks for the bug report!

**Comment by zoe1337** [comment]
thanks. I also had issues when trying to build from a dedicated build directory. simply `mkdir build; cd build; configure ..; make` will fail
I should open up a new issue for that, I guess

**Comment by ossama-othman** [comment]
The dedicated build directory problem (`VPATH` build) was fixed by pull request #73 that was recently merged to the `master` branch.  Please give it a try if you get a chance.

I'm still reworking the plugin test infrastructure to address the original problem you reported.

**Comment by ossama-othman** [comment]
[mptcpd 0.5](https://github.com/intel/mptcpd/releases/tag/v0.5) was just released.  It fixes this problem.
